src/main/scala/io/conduktor/ksm/source/HttpSourceAcl.scala

```diff
@@ -0,0 +1,216 @@
+    * Ideally this function is smart and does not pull the entire external Acl at every iteration
+    * Return `None` if the Source ACLs have not changed (usually using metadata).
+    * Return `Some(x)` if the ACLs have changed. `x` represents the parsing and parsing errors if any
+    * Note: the first call to this function should never return `None`.
+    *
+    * Kafka Security Manager will not update ACLs in Kafka until there are no errors in the result
+    *
+    * @return
+    */
+  override def refresh(): Option[ParsingContext] = {
+
+    val request: Request = new Request(uri)
+    // super important in order to properly fail in case a timeout happens for example
+    request.enableThrowingIOException(true)
+    request.followRedirects(false)
+    request.connectTimeoutMillis(Int.MaxValue)
+    request.readTimeoutMillis(Int.MaxValue)
+
+    if (authenticationType == AuthenticationType.GOOGLE_IAM) {
+      request.header("Authorization", getIdToken())
+    }
+
+    // we use this header for the 304
+    lastModified.foreach(header => request.header("If-Modified-Since", header))
+    request.header("Content-Type", "text/plain") // only type supported for now
+    val response: Response = HTTP.request(httpMethod, request)
+
+    response.status match {
+      case 200 =>
+        lastModified = response.header("Last-Modified")
+
+        validateBodyLength(response)
+
+        Some(
+          ParsingContext(
+            parserRegistry.getParser(this.parser),
+            new BufferedReader(new StringReader(response.textBody))
+          )
+        )
+      case 304 =>
+        None
+      case _ =>
+        // we got an http error so we propagate it
+        log.error("HTTP Status {} in HTTP Source request: {}", response.status, response.asString)
+        throw HTTPException(Some(response.asString), response)
+    }
+  }
+
+  private def validateBodyLength(response: Response): Unit = {
+    val optContentLengthHeader = response.headers
+      .find(h => CONTENT_LENGTH.equalsIgnoreCase(h._1))
+      .map(h => h._2)
+      .map(l => l.toInt)
+    if (optContentLengthHeader.isEmpty) {
+      log.warn(s"Response doesn't contain $CONTENT_LENGTH header, only contained the following headers: ${response.headers.keySet}. Skipping validation...")
+      return
+    }
+
+    val contentLengthHeader = optContentLengthHeader.get
+    val bodyLength = response.asBytes.length
+    log.info(s"Validating body length ($bodyLength bytes) received from $uri against Content-Length header ($contentLengthHeader bytes) claimed in response")
+    try {
+      val reader = new BufferedReader(new StringReader(response.textBody))
+      val aclLineCount = reader.lines.count
+      log.info("There were {} lines in the response received from {}", aclLineCount, uri)
+      reader.close()
+    }
+    catch {
+      case e: Exception => log.warn(s"Failed to compute number of lines in the response received from $uri", e)
+    }
```

Comment:
logging body length and number of lines is not so useful to me

Reply:
As we faced some errors in production regarding this, we included this on log to be sure if it was a problem in our backend or in skinny HTTP client. 
Can I change to trace level at least?

Reply:
fine. but this can be simplified:
```
    log.debug("There were {} lines in the response received from {}", response.textBody.lines().count(), uri)
```

Reply:
also can you add a comment that skinny http does not check content length so it  must be done manually, Thanks

Reply:
Done